### PR TITLE
HIVE-11417. Move the ReaderImpl and the RecordReaderImpl to the ORC module.

### PR DESCRIPTION
**Pull request by omalley**

### CODE REVIEW

File: orc/src/java/org/apache/orc/impl/HadoopShims.java [hunk 1]
```diff
@@ -43,6 +46,75 @@
...
+   * Read data into a Text object in the fastest way possible
+   */
+  public interface TextReaderShim {
+    /**
+     * @param txt
+     * @param size
+     * @return bytes read
+     * @throws IOException
+     */
+    void read(Text txt, int size) throws IOException;
```
Comment (prasanthj) [inline orc/src/java/org/apache/orc/impl/HadoopShims.java]: Is there a minimum supported hadoop version for orc? If we don't support older hadoop version then this shim might not be required.

File: orc/src/java/org/apache/orc/impl/HadoopShimsCurrent.java [hunk 2]
```diff
@@ -59,4 +63,30 @@ public DirectDecompressor getDirectDecompressor(
...
+                                              ByteBufferPoolShim pool
+                                              ) throws IOException {
+    return ZeroCopyShims.getZeroCopyReader(in, pool);
+  }
+
+  private final class FastTextReaderShim implements TextReaderShim {
+    private final DataInputStream din;
+
+    public FastTextReaderShim(InputStream in) {
+      this.din = new DataInputStream(in);
```
Comment (prasanthj) [inline orc/src/java/org/apache/orc/impl/HadoopShimsCurrent.java]: Where do we close this stream?

File: orc/src/java/org/apache/orc/impl/OrcAcidUtils.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  }
+
+  private static final Charset utf8 = Charset.forName("UTF-8");
+  private static final CharsetDecoder utf8Decoder = utf8.newDecoder();
+
+  public static AcidStats parseAcidStats(Reader reader) {
+    if (reader.hasMetadataValue(ACID_STATS)) {
+      String statsSerialized;
+      try {
+        ByteBuffer val =
```
Comment (prasanthj) [inline orc/src/java/org/apache/orc/impl/OrcAcidUtils.java]: nit: bad alignment

File: orc/src/java/org/apache/orc/impl/ReaderImpl.java [hunk 4]
```diff
@@ -0,0 +1,758 @@
...
+import org.apache.orc.StripeInformation;
+import org.apache.orc.StripeStatistics;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.hive.common.io.DiskRange;
+import org.apache.hadoop.hive.ql.util.JavaDataModel;
```
Comment (prasanthj) [inline orc/src/java/org/apache/orc/impl/ReaderImpl.java]: ql dependency?

Comment (prasanthj) [inline orc/src/java/org/apache/orc/impl/ReaderImpl.java]: Can this be moved to common instead?

Comment (prasanthj) [inline orc/src/java/org/apache/orc/impl/ReaderImpl.java]: Never mind. This is already part of storage-api. So not a problem.

File: orc/src/java/org/apache/orc/impl/HadoopShims.java (hunk 1 above)
Comment (omalley) [inline orc/src/java/org/apache/orc/impl/HadoopShims.java]: For now we should stick with what Hive supports. I copied this code directly from Hive's HadoopShims.

File: orc/src/java/org/apache/orc/impl/OrcAcidUtils.java (hunk 3 above)
Comment (omalley) [inline orc/src/java/org/apache/orc/impl/OrcAcidUtils.java]: I'm not sure what you think is misaligned, but I simplified this code.

File: orc/src/java/org/apache/orc/impl/HadoopShimsCurrent.java (hunk 2 above)
Comment (omalley) [inline orc/src/java/org/apache/orc/impl/HadoopShimsCurrent.java]: I copied this code from Hive's HadoopShims, but yes we should close the stream.

Comment (omalley) [inline orc/src/java/org/apache/orc/impl/HadoopShimsCurrent.java]: Ok, the file stream is the same one as the DefaultDataReader, but that one is also not closed. I'll make the zerocopyshim closeable.

**Pull request closed without merging**